src/snmalloc/mem/corealloc.h

```diff
@@ -885,6 +886,10 @@ namespace snmalloc
     {
       size_t rsize = sizeclass_to_size(sizeclass);
 
+      void* result = SecondaryAllocator::allocate(rsize);
+      if (result != nullptr)
+        return capptr::Alloc<void>::unsafe_from(result);
+
```

Comment:
Thinking about this a bit more.  I wonder if this should be at the start of `small_alloc`.  I think there are allocation patterns that would never hit this path. It is possible to work completely without having to grab a new chunk from the backend.

Sorry, I think I had this wrong in my original issue.